packetbeat/protos/memcache: panics when a transaction is empty

This has been observed for 7.13.0, but the code has not been changed since in the relevant path.
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x220 pc=0x5645e4988659]

goroutine 12122 [running]:
github.com/elastic/beats/v7/packetbeat/protos/memcache.(*transaction).Event(0x0, 0xc000475ed0, 0x5645e498d353, 0x5645e719a12b)
        /go/src/github.com/elastic/beats/packetbeat/protos/memcache/memcache.go:387 +0x39
github.com/elastic/beats/v7/packetbeat/protos/memcache.(*memcache).onTransaction(0xc00149be10, 0x0)
        /go/src/github.com/elastic/beats/packetbeat/protos/memcache/memcache.go:207 +0x71
github.com/elastic/beats/v7/packetbeat/protos/memcache.(*memcache).finishTransaction(...)
        /go/src/github.com/elastic/beats/packetbeat/protos/memcache/memcache.go:199
github.com/elastic/beats/v7/packetbeat/protos/memcache.(*memcache).onUDPTrans(0xc00149be10, 0xc002ff49c0, 0x0, 0x0)
        /go/src/github.com/elastic/beats/packetbeat/protos/memcache/plugin_udp.go:213 +0x96
github.com/elastic/beats/v7/packetbeat/protos/memcache.(*memcache).ParseUDP.func1()
        /go/src/github.com/elastic/beats/packetbeat/protos/memcache/plugin_udp.go:148 +0x7e
created by time.goFunc
        /usr/local/go/src/time/sleep.go:167 +0x46
```

This happens when a `*transaction` is constructed from a UDP stream, but has no request and no response. In this case the `*transaction` is nil by the time it gets to `*memcache.onTransaction`. In `*memcache.Event`, the `*transaction.Notes` field is accessed, dereferencing the nil pointer and Bang! There is a panic recover in the `ParseUDP` which looks to have been intended to recover from events like this, but the panic happens in a separate goroutine in a `time.AfterFunc` call, so the recover cannot see it.

For educational purposes, this is the minimal reproducer: https://play.golang.com/p/6oHEQl9vkoV

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -100,6 +100,7 @@ https://github.com/elastic/beats/compare/v8.2.0\...main[Check the HEAD diff]
 
 *Packetbeat*
 
+- Fix panic on memcache transaction with no request or response. {issue}33852[33852] {pull}33853[33853]
 
 *Winlogbeat*
 

--- a/packetbeat/protos/memcache/memcache.go
+++ b/packetbeat/protos/memcache/memcache.go
@@ -198,6 +198,9 @@ func (mc *memcache) GetPorts() []int {
 }
 
 func (mc *memcache) finishTransaction(t *transaction) error {
+	if t == nil {
+		return nil
+	}
 	mc.handler.onTransaction(t)
 	return nil
 }

--- a/packetbeat/protos/memcache/plugin_udp.go
+++ b/packetbeat/protos/memcache/plugin_udp.go
@@ -144,6 +144,7 @@ func (mc *memcache) ParseUDP(pkt *protos.Packet) {
 	}
 	if !done {
 		trans.timer = time.AfterFunc(mc.udpConfig.transTimeout, func() {
+			defer logp.Recover("ParseMemcache(UDP) panic during forward")
 			debug("transaction timeout -> forward")
 			mc.onUDPTrans(trans)
 			mc.udpExpTrans.push(trans)
@@ -261,6 +262,9 @@ func (c *udpConnection) killTransaction(t *udpTransaction) {
 }
 
 func (lst *udpExpTransList) push(t *udpTransaction) {
+	if t == nil {
+		return
+	}
 	lst.Lock()
 	defer lst.Unlock()
 	t.next = lst.head

